pkg/kubelet/lifecycle/handlers.go

```diff
@@ -128,8 +155,20 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 			return "", err
 		}
 	}
-	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
-	resp, err := hr.httpGetter.Get(url)
+	path := handler.HTTPGet.Path
+	scheme := "http"
+	if string(handler.HTTPGet.Scheme) != "" {
```

Comment:
prefer comparison to the declared constants over string casting them to be the scheme used in the URL:

```
switch handler.HTTPGet.Scheme {
case URISchemeHTTPS:
  scheme = "https"
default:
  scheme = "http"
}
```